### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/boostorg-optional/commit/f7b7a1442aafd1df1658b2dd949cfcd76d5ddd94#commitcomment-88602764

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/boostorg-optional/commit/5042fd201bf360db409812ae56319f15108350a3

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: doc/28_ref_optional_semantics.qbk [hunk 1]
```diff
@@ -163,7 +163,7 @@ assert (!uninit);
 optional<std::unique_ptr<T>> uinit2 ( std::move(uninit) ) ;
 assert ( uninit2 == uninit );
 
-optional<std::unique_ptr<T>> init( std::uniqye_ptr<T>(new T(2)) );
+optional<std::unique_ptr<T>> init( std::unique_ptr<T>(new T(2)) );
```
Comment (jsoref) [inline doc/28_ref_optional_semantics.qbk:166]: It's possible this is intentional...

File: include/boost/optional/detail/optional_reference_spec.hpp [hunk 2]
```diff
@@ -130,7 +130,7 @@ class optional<T&> : public optional_detail::optional_tag
         explicit optional(const optional<U&>& rhs) BOOST_NOEXCEPT : ptr_(rhs.get_ptr()) {}
     optional(const optional& rhs) BOOST_NOEXCEPT : ptr_(rhs.get_ptr()) {}
     
-    // the following two implement a 'conditionally explicit' constructor: condition is a hack for buggy compilers with srewed conversion construction from const int
+    // the following two implement a 'conditionally explicit' constructor: condition is a hack for buggy compilers with screwed conversion construction from const int
```
Comment (jsoref) [inline include/boost/optional/detail/optional_reference_spec.hpp:133]: ?

**Review by jsoref — COMMENTED** [review]
Most corrections automatically suggested by Google Sheets. All fault mine

**Comment by akrzemi1** [comment]
Thanks for the fix.

**Pull request merged**